### PR TITLE
cache nodeName once nodeName is returned by getNodeName()  rather than regenerate it

### PR DESCRIPTION
**Pull request by NickrenREN**

### CODE REVIEW

**Comment by resouer** [comment]
@k8s-bot ok to test

**Comment by NickrenREN** [comment]
@resouer  Thanks but testing doesn't start

**Comment by resouer** [comment]
@k8s-bot ok to test

**Comment by NickrenREN** [comment]
@k8s-bot unit test this

**Comment by NickrenREN** [comment]
Hi @resouer ,  do you have time to help with the review? Thanks

**Comment by resouer** [comment]
Yes, will do review soon.

File: cmd/kubelet/app/server.go [hunk 1]
```diff
@@ -379,11 +380,12 @@ func run(s *options.KubeletServer, kubeDeps *kubelet.KubeletDeps) (err error) {
 			}
 		}
 
+		nodeName, err = getNodeName(cloud, nodeutil.GetHostname(s.HostnameOverride))
+		if err != nil {
+			return err
+		}
+
 		if s.BootstrapKubeconfig != "" {
-			nodeName, err := getNodeName(cloud, nodeutil.GetHostname(s.HostnameOverride))
```
Comment (resouer) [inline cmd/kubelet/app/server.go]: why move this part out of `s.BootstrapKubeconfig` check?

Comment (NickrenREN) [inline cmd/kubelet/app/server.go]: Oh,I just think we can get nodeName first and then use nodeName somewhere, here use nodeName in s.BootstrapKubeconfig != "" check and maybe somewhere else.
I can move it back if my understanding is wrong.

Comment (resouer) [inline cmd/kubelet/app/server.go]: I would suggest to keep this not changed.

Comment (NickrenREN) [inline cmd/kubelet/app/server.go]: Ok, thanks @resouer . I will move it back tomorrow.

**Comment by NickrenREN** [comment]
@resouer  I have moved that back. PTAL. Thanks

File: cmd/kubelet/app/server.go (hunk 1 above)
Comment (NickrenREN) [inline cmd/kubelet/app/server.go]: @resouer I have moved that back. PTAL. Thanks

**Comment by resouer** [comment]
@derekwaynecarr Please have a look and approve

**Comment by NickrenREN** [comment]
@derekwaynecarr   @vishh  @yujuhong  PTAL. Thanks

**Comment by NickrenREN** [comment]
@k8s-bot non-cri e2e test this

**Comment by NickrenREN** [comment]
@vishh  @derekwaynecarr   Could you help add approve label? Thanks

**Comment by NickrenREN** [comment]
@resouer Complete the rebase. PTAL

**Comment by NickrenREN** [comment]
@k8s-bot cvm gce e2e test this

**Comment by NickrenREN** [comment]
Just do the rebase.

**Comment by NickrenREN** [comment]
Finish the rebase.  @resouer @yujuhong  ptal

**Comment by yujuhong** [comment]
@NickrenREN how costly is it to regenerate the node name? Is this PR fixing a bug or just trying to optimize code?

**Comment by NickrenREN** [comment]
just try to optimize code

**Comment by mtaufen** [comment]
@NickrenREN are you planning to rebase this?

**Comment by NickrenREN** [comment]
Yeah

**Pull request closed without merging**